src/server/pfs/s3/multipart.go

```diff
@@ -80,24 +80,15 @@ type multipartController struct {
 	maxAllowedParts int
 }
 
-func newMultipartController(pc *client.APIClient, logger *logrus.Entry, repo string, maxAllowedParts int) *multipartController {
-	return &multipartController{
-		pc:              pc,
-		logger:          logger,
-		repo:            repo,
-		maxAllowedParts: maxAllowedParts,
-	}
-}
-
-func (c *multipartController) ensureRepo() error {
-	_, err := c.pc.InspectBranch(c.repo, "master")
+func (c multipartController) ensureRepo(pc *client.APIClient) error {
+	_, err := pc.InspectBranch(c.repo, "master")
 	if err != nil {
-		err = c.pc.CreateRepo(c.repo)
+		err = pc.CreateRepo(c.repo)
```

Comment:
This might be a tiny bit cleaner if you set the `Update` field in the `CreateRepoRequest` it implements upsert semantics so you won't have to check error messages.